tools/build_file_generator/BuildFileGenerator.java

```diff
@@ -276,25 +345,113 @@ private CheckstyleTestExtension getCheckstyleTest(String target) {
                 .get(0)
                 .getTarget();
       }
-      if (!result.contains(target)) {
+      if (target.equals("//third_party/maven/com/google/dagger:dagger")
+          || target.equals("//common:dagger_common_component")) {
+        target = "//common:dagger_with_annotation_processor";
+      }
+      if (target.equals("//common/repo:git_repo_factory")) {
```

Comment:
Please add a TODO to generalize this. Is `//common/repo:repo` BUILD manually generated?
Because if we had `//common/repo:git_repo` then we could generalize to remove `_factory`.

Reply:
`//common/repo:repo` BUILD is generated manually at the moment. If I replace it with autogenerated one `//common/repo:git_repo` doesn't work. For some reason, `GitRepoFactory` isn't available in the code.

Reply:
Ok, thanks.